### DISCUSSION TITLE
How to have proper type casting for custom log level?

### DISCUSSION THREAD

**Discussion by dhruvinsh** [microsoft/pyright#5418] [Q&A] [answered]
I have implemented non-standard logging method like `success` and `fail`

Now pyright complains about, `cannot access member "fail" for type "Logger" Member "fail" is unknow`

How can I config pyright such a way that it can recognize the method as valid method?

```python
def create_logger(name: Optional[str] = None):
    """
    create logger object
    """
    # setup logger to custom logger
    logging.setLoggerClass(CustomLogger)
    return logging.getLogger(name or __name__)
```

**Comment by debonte** [comment, marked as answer]
We rely on the typeshed stubs to describe how these APIs work. The stub for `getLogger` looks like this:

```python
def getLogger(name: str | None = None) -> Logger: ...
```

So Pyright expects that the object returned by `logging.getLogger` and `create_logger` will be of type `Logger` rather than `CustomLogger`. One approach to informing Pyright of your expectations would be to `cast` that return value:

```python
return cast(CustomLogger, logging.getLogger(name or __name__))
```

**Reply by dhruvinsh** [reply]
Thank you for your answer, I will give this a try and mark it if that solves my problem. I totally forgot about `typing.cast` but this could be it. 

Update: above solution worked. Thank you so much.
